src/PIL/Image.py

```diff
@@ -1364,7 +1370,7 @@ def getcolors(self, maxcolors=256):
             return out
         return self.im.getcolors(maxcolors)
 
-    def getdata(self, band=None):
+    def getdata(self, band: int | None = None) -> Image:
```

Comment:
This returns an ImagingCore instance, not Image.

Reply:
```suggestion
    def getdata(self, band: int | None = None):
```